src/components/colorscale/defaults.js

```diff
@@ -15,20 +15,63 @@ var hasColorbar = require('../colorbar/has_colorbar');
 var colorbarDefaults = require('../colorbar/defaults');
 
 var isValidScale = require('./scales').isValid;
+var traceIs = require('../../registry').traceIs;
 
-function npMaybe(cont, prefix) {
+function npMaybe(outerCont, prefix) {
     var containerStr = prefix.slice(0, prefix.length - 1);
     return prefix ?
-        Lib.nestedProperty(cont, containerStr).get() || {} :
-        cont;
+        Lib.nestedProperty(outerCont, containerStr).get() || {} :
+        outerCont;
 }
 
-module.exports = function colorScaleDefaults(traceIn, traceOut, layout, coerce, opts) {
+module.exports = function colorScaleDefaults(outerContIn, outerContOut, layout, coerce, opts) {
```

Comment:
Having two `outs` this may be a bit difficult to read. Could we replace `outer` with something like `parent` or `ext` instead?

Reply:
Good call. `parentContIn` and `parentContOut` ok with you?

Reply:
Yes please that's OK. 
It could also be applied in `npMaybe` function above.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3803/commits/29a92a60ad17c2dfcae1021a924f3e2f433058b2